### PR TITLE
DataTable: Non-Lazy: integrationtests filterMatchModes (including bug fix(es))

### LINKED ISSUE #7715: DataTable: FilterMatchMode gte is broken
Current impl of GreaterThanEqualsFilterConstraint is `return (o1, o2) -> o1.compareTo(o2) == 0;`

Fix comes as part of a larger Integrationtest-PR.

### PR DESCRIPTION
**Pull request by christophs78**

### CODE REVIEW

File: primefaces-integration-tests/src/test/java/org/primefaces/integrationtests/datatable/DataTable026Test.java [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+                .filter(e -> e.getBirthDate().isAfter(LocalDate.of(1969, 12, 24)) && e.getBirthDate().isBefore(LocalDate.of(1970,1, 4)))
+                .collect(Collectors.toList());
+        assertEmployeeRows(dataTable, employeesFiltered);
+
+        // Setting range via Selenium (sendKeys) causes JS-errors. So we don´t check for them.
+//        assertConfiguration(dataTable.getWidgetConfiguration());
+    }
+
+    private void assertConfiguration(JSONObject cfg) {
+        assertNoJavascriptErrors();
```
Comment (christophs78) [inline primefaces-integration-tests/src/test/java/org/primefaces/integrationtests/datatable/DataTable026Test.java:305]: Maybe we should add this to assertNoJavascriptErrors
https://stackoverflow.com/questions/51404360/how-to-clear-console-errors-using-selenium

Comment (melloware) [reply primefaces-integration-tests/src/test/java/org/primefaces/integrationtests/datatable/DataTable026Test.java:305]: Yes we should add a `clearConsoleErrors()` method and add it.

**Review by melloware — APPROVED** [review]

File: primefaces-selenium/primefaces-selenium-core/src/main/java/org/primefaces/selenium/AbstractPrimePageTest.java [hunk 2]
```diff
@@ -47,13 +38,25 @@
...
 @TestInstance(TestInstance.Lifecycle.PER_CLASS)
 @TestMethodOrder(MethodOrderer.OrderAnnotation.class)
 @ExtendWith(BootstrapExtension.class)
 @ExtendWith(WebDriverExtension.class)
 @ExtendWith(PageInjectionExtension.class)
 public abstract class AbstractPrimePageTest {
 
+    @BeforeEach
+    public void beforeEach() {
+        clearConsole();
```
Comment (christophs78) [inline primefaces-selenium/primefaces-selenium-core/src/main/java/org/primefaces/selenium/AbstractPrimePageTest.java:57]: added clearConsole - method and call it before each test

File: primefaces-selenium/primefaces-selenium-core/src/main/java/org/primefaces/selenium/AbstractPrimePageTest.java [hunk 3]
```diff
@@ -192,6 +195,17 @@ protected void assertNoJavascriptErrors() {
         Assertions.assertTrue(severe.isEmpty(), "Javascript errors were detected in the browser console.\r\n" + severe.toString());
     }
 
+    /**
+     * Clears the browser console.
+     */
+    protected void clearConsole() {
+        // https://stackoverflow.com/questions/51404360/how-to-clear-console-errors-using-selenium
+        JavascriptExecutor js = (JavascriptExecutor) getWebDriver();
```
Comment (tandraschko) [inline primefaces-selenium/primefaces-selenium-core/src/main/java/org/primefaces/selenium/AbstractPrimePageTest.java:203]: @christophs78 just one thing: dont we have something like PrimeSelenium.executeScript?

**Pull request merged**